src/audio/dai-zephyr.c

```diff
@@ -415,8 +401,13 @@ static int dai_comp_hw_params(struct comp_dev *dev, struct sof_ipc_stream_params
 static int dai_verify_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
 {
 	struct sof_ipc_stream_params hw_params;
+	int ret;
 
-	dai_comp_get_hw_params(dev, &hw_params, params->direction);
+	ret = dai_comp_get_hw_params(dev, &hw_params, params->direction);
```

Comment:
why not remove dai_get_hw_params() altogether and just set the params here?

Reply:
yes, this is also achievable, if there is no other concern and history, I can remove dai_get_hw_params totally.
@ranj063 @lgirdwood

Reply:
it is a static function. so I dont see any concern with removing it. That function doesnt really get anything but rather just set things. So might as well do those things directly here.

Reply:
ok, done